### PR TITLE
feat: modify position

### PR DESCRIPTION
**Pull request by dannydaniil**

### CODE REVIEW

File: v3-sdk/modifying-position/src/config.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import { Token } from '@uniswap/sdk-core'
+import { FeeAmount } from '@uniswap/v3-sdk'
+import { DAI_TOKEN, USDC_TOKEN } from './libs/constants'
+
+// Sets if the example should run locally or on chain
+export enum Environment {
+  LOCAL,
+  WALLET_EXTENSION,
+  PRODUCTION,
```
Comment (adjkant) [inline v3-sdk/modifying-position/src/config.ts:9]: Should change to `MAINNET` per #8

File: v3-sdk/modifying-position/src/example/Example.css [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+.App-header {
+  background-color: #282c34;
+  min-height: 100vh;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  font-size: calc(10px + 2vmin);
+  color: white;
+}
```
Comment (adjkant) [inline v3-sdk/modifying-position/src/example/Example.css:14]: See CSS changes in `#8` too for simplification

File: v3-sdk/modifying-position/src/example/Example.tsx [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+}
+
+const getPosition = async (
+  percentageToIncrease?: number
+): Promise<Position> => {
+  // get pool info
+  const poolInfo = await getPoolInfo()
+
+  // construct pool instance
+  const USDC_DAI_POOL = new Pool(
```
Comment (adjkant) [inline v3-sdk/modifying-position/src/example/Example.tsx:90]: Do we want to break this into the `positions.ts` file now or in a followup when we do this for others?

File: v3-sdk/modifying-position/src/example/Example.tsx [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+
+  // get calldata for minting a position
+  const { calldata, value } = NonfungiblePositionManager.addCallParameters(
+    await getPosition(),
+    {
+      recipient: address,
+      deadline: Math.floor(Date.now() / 1000) + 60 * 20,
+      slippageTolerance: new Percent(50, 10_000),
+    }
+  )
```
Comment (adjkant) [inline v3-sdk/modifying-position/src/example/Example.tsx:163]: nit: breaking this down here doesn't show the intermediary type / fetching well. Thoughts on giving this a name and using below instead? Can also add a type to show what the result is etc

File: v3-sdk/modifying-position/src/example/Example.tsx [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+  }
+
+  // get calldata for minting a position
+  const { calldata, value } = NonfungiblePositionManager.addCallParameters(
+    await getPosition(),
+    {
+      recipient: address,
+      deadline: Math.floor(Date.now() / 1000) + 60 * 20,
+      slippageTolerance: new Percent(50, 10_000),
+    }
```
Comment (adjkant) [inline v3-sdk/modifying-position/src/example/Example.tsx:162]: Define this on its own to show the type / make this section smaller?

File: v3-sdk/modifying-position/src/example/Example.tsx [hunk 6]
```diff
@@ -0,0 +1,386 @@
...
+      deadline: Math.floor(Date.now() / 1000) + 60 * 20,
+      slippageTolerance: new Percent(50, 10_000),
+      tokenId: positionId,
+      // percentage of liquidity to remove
+      liquidityPercentage: new Percent(CurrentConfig.tokens.percentageToRemove),
+      collectOptions: {
+        expectedCurrencyOwed0: CurrencyAmount.fromRawAmount(DAI_TOKEN, 0),
+        expectedCurrencyOwed1: CurrencyAmount.fromRawAmount(USDC_TOKEN, 0),
+        recipient: address,
+      },
```
Comment (adjkant) [inline v3-sdk/modifying-position/src/example/Example.tsx:236]: Make its own const?

File: v3-sdk/modifying-position/src/example/Example.tsx [hunk 7]
```diff
@@ -0,0 +1,386 @@
...
+    const subscription = getProvider()?.on('block', callback)
+    return () => {
+      subscription?.removeAllListeners()
+    }
+  })
+}
+
+const Example = () => {
+  const [tokenInBalance, setTokenInBalance] = useState<string>()
+  const [tokenOutBalance, setTokenOutBalance] = useState<string>()
```
Comment (adjkant) [inline v3-sdk/modifying-position/src/example/Example.tsx:265]: TokenIn and TokenOut here, should remove for consistency to token0 and token1

File: v3-sdk/modifying-position/src/libs/constants.ts [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  18,
+  'DAI',
+  'Dai Stablecoin'
+)
+
+// Transactions
+
+export const MAX_FEE_PER_GAS = '100000000000'
+export const MAX_PRIORITY_FEE_PER_GAS = '100000000000'
+export const AMOUNT_TO_APPROVE = 1000000000000
```
Comment (adjkant) [inline v3-sdk/modifying-position/src/libs/constants.ts:34]: more specific name here? Unclear when not being used in context

File: v3-sdk/modifying-position/src/libs/conversion.ts [hunk 9]
```diff
@@ -0,0 +1,7 @@
+export function fromReadableAmount(amount: number, decimals: number): number {
+  return amount * Math.pow(10, decimals)
+}
+
+export function toReadableAmount(rawAmount: number, decimals: number): number {
+  return rawAmount / Math.pow(10, decimals)
+}
```
Comment (adjkant) [inline v3-sdk/modifying-position/src/libs/conversion.ts:7]: Should grab the latest to avoid under/overflow issues

**Review by adjkant — COMMENTED** [review]
Some org and naming but the core is good!

File: v3-sdk/modifying-position/src/example/Example.tsx (hunk 3 above)
Comment (dannydaniil) [reply v3-sdk/modifying-position/src/example/Example.tsx:90]: I can address this now :)

File: v3-sdk/modifying-position/src/example/Example.tsx (hunk 5 above)
Comment (dannydaniil) [reply v3-sdk/modifying-position/src/example/Example.tsx:162]: Sounds good! I can do it for minting too

File: v3-sdk/modifying-position/src/libs/conversion.ts (hunk 9 above)
Comment (dannydaniil) [reply v3-sdk/modifying-position/src/libs/conversion.ts:7]: Good call - I wanted to that and I forgot!

Comment (dannydaniil) [reply v3-sdk/modifying-position/src/libs/conversion.ts]: Leaving this out for now - number are too small

File: v3-sdk/modifying-position/src/example/Example.tsx [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+} from '../libs/providers'
+import {
+  MAX_FEE_PER_GAS,
+  MAX_PRIORITY_FEE_PER_GAS,
+  NONFUNGIBLE_POSITION_MANAGER_CONTRACT_ADDRESS,
+  DAI_TOKEN,
+  USDC_TOKEN,
+} from '../libs/constants'
+
+async function mintPosition(): Promise<TransactionState> {
```
Comment (adjkant) [inline v3-sdk/modifying-position/src/example/Example.tsx:33]: move to `positions.ts`, potentially rename to `liquidity.ts`?

**Review by adjkant — DISMISSED** [review]

**Pull request merged**